traffic_ops/traffic_ops_golang/cachegroupparameter/parameters.go

```diff
@@ -180,3 +185,124 @@ func (cgparam *TOCacheGroupParameter) Delete() (error, error, int) {
 
 	return api.GenericDelete(cgparam)
 }
+
+// ReadAllCacheGroupParameters reads all cachegroup parameter associations.
+func ReadAllCacheGroupParameters(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+	output, err := GetAllCacheGroupParameters(inf.Tx.Tx)
+	if err != nil {
+		api.WriteRespAlertObj(w, r, tc.ErrorLevel, "querying cachegroupparameters with error: "+err.Error(), output)
+		return
+	}
+	api.WriteResp(w, r, output)
+}
+
+// GetAllCacheGroupParameters gets all cachegroup associations from the database and returns as slice.
+func GetAllCacheGroupParameters(tx *sql.Tx) (tc.CacheGroupParametersList, error) {
+	parameters := map[string]string{
+		"orderby": "cachegroup",
```

Comment:
why the change from https://github.com/apache/trafficcontrol/blob/f1447bde94e84aa93fba9506a387a34e9aea3dd7/traffic_ops/traffic_ops_golang/cachegroupparameter/parameters.go#L210 also the parameters passed into the `BuildWhereAndOrderByAndPagination` should be parameters that the APIInfo parsed out ```inf.Params```

Reply:
i was thinking to mirror the perl endpoint better but added it back in as well as docs

Reply:
Lemme throw this out there: you should support a proper superset of the query parameters supported by Perl - more is totally acceptable, if you want, as long as they are documented (along with the version in which they were added).